### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by irynamekh**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    build: .
+    env_file: ./.env
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysqldb:$MYSQL_DOCKER_PORT/$MYSQL_DB
+      SPRING_DATASOURCE_USERNAME: $MYSQL_USER
+      SPRING_DATASOURCE_PASSWORD: $MYSQL_PASSWORD
+      JAVA_TOOL_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:$DEBUG_PORT
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
```
Comment (JeniaSan) [inline docker-compose.yaml:32]: add empty line to the end of the file)

File: src/main/java/com/bookstore/controller/OrderController.java [hunk 2]
```diff
@@ -53,8 +58,9 @@ public OrderItemResponseDto getOrderItem(@PathVariable Long orderId,
     @Operation(summary = "Place an order",
             description = "Create a new order")
     public OrderResponseDto createOrder(
-            @RequestBody @Valid CreateOrderRequestDto requestDto) {
-        return orderService.save(requestDto);
+            @RequestBody @Valid CreateOrderRequestDto requestDto, Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.save(requestDto, user.getId());
```
Comment (JeniaSan) [inline src/main/java/com/bookstore/controller/OrderController.java:63]: You can do transformation logic in the service. 
```suggestion
        return orderService.save(requestDto, authentication);
```

**Review by JeniaSan — APPROVED** [review]
Good job, but I left some comments

File: .env [hunk 3]
```diff
@@ -0,0 +1,9 @@
+MYSQL_ROOT_PASSWORD=RootPassword
+MYSQL_USER=admin
+MYSQL_PASSWORD=Password123
+MYSQL_DB=book_store
+MYSQL_LOCAL_PORT=3307
+MYSQL_DOCKER_PORT=3306
+SPRING_LOCAL_PORT=8087
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (nacenik) [inline .env]: empty line

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# Final stage
+FROM openjdk:17-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.JarLauncher"]
+EXPOSE 8080
```
Comment (nacenik) [inline Dockerfile]: empty line

**Review by nacenik — CHANGES_REQUESTED** [review]
GJ, Let's add some improvements

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**